sec1_intro_and_setup/test_app/app/views/layouts/application.html.erb

```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
```

Comment:
Rails will use the layout defined in the file `app/views/layouts/application.html.erb` as a default for rendering any page. 

- The `viewport` metatag improves the presentation of web pages on mobile devices. Setting a viewport tells the browser how content should fit on the device’s screen.
- The `stylesheet_link_tag` and `javascript_include_tag` elements add CSS stylesheets and JavaScript from the Rails asset pipeline.
>(Note that: The asset pipeline provides a framework to concatenate and minify or compress JavaScript and CSS assets. It also adds the ability to write these assets in other languages and pre-processors such as CoffeeScript, Sass, and ERB. It allows assets in your application to be automatically combined with assets from other gems.)

- The `data-turbolinks-track` attribute supports Rails Turbolinks. 
> (Note that: Turbolinks makes following links in your web application faster. Instead of letting the browser recompile the JavaScript and CSS between each page change, it keeps the current page instance alive and replaces only the body (or parts of) and the title in the head.)